### PR TITLE
OpenBSD: add `sendmmsg` and `recvmmsg` support

### LINKED ISSUE #3696: Add `recvmmsg` and `sendmmsg` on OpenBSD (*-unknown-openbsd)
`recvmmsg` is supported on OpenBSD since OpenBSD 7.2: https://www.openbsd.org/72.html
Same for `sendmmsg`.

Documentation is here:
- https://man.openbsd.org/OpenBSD-7.2/recvmmsg
- https://man.openbsd.org/OpenBSD-7.2/sendmmsg

Related PR made at the time when OpenBSD did not support these functions yet: https://github.com/rust-lang/libc/pull/1231

### PR DESCRIPTION
**Pull request by devnexen**
close #3696

### CODE REVIEW

**Comment by tgross35** [comment]
Changes lgtm, can you target `main`?

@rustbot label +stable-nominated

File: src/unix/bsd/netbsdlike/mod.rs [hunk 1]
```diff
@@ -837,6 +837,19 @@ extern "C" {
...
     ) -> ::c_int;
+    pub fn sendmmsg(
+        sockfd: ::c_int,
+        msgvec: *mut ::mmsghdr,
+        vlen: ::c_uint,
+        flags: ::c_int,
+    ) -> ::c_int;
+    pub fn recvmmsg(
+        sockfd: ::c_int,
+        msgvec: *mut ::mmsghdr,
```
Comment (tgross35) [inline src/unix/bsd/netbsdlike/mod.rs:848]: `msgvec` is `mmsghdr` per https://man.openbsd.org/sendmsg.2 and https://man.openbsd.org/recvmsg.2

**Comment by tgross35** [comment]
I was wondering why this can't just be combined for all BSD, but answered my own question - FreeBSD uses `size_t`, OpenBSD uses `unsigned int`.

- https://man.freebsd.org/cgi/man.cgi?query=sendmmsg&manpath=FreeBSD+11.0-RELEASE
- https://man.openbsd.org/sendmsg.2

**Comment by devnexen** [comment]
closed in favor of #3831

**Pull request closed without merging**